project/src/main/ui/menu/practice-menu.gd

```diff
@@ -1,147 +1,202 @@
+	# load the player's previously played level
+	_level_settings.load_from_resource(PlayerData.practice.level_id)
+
+
+## Enables/disables focus for inputs as submenus are shown and hidden.
+##
+## When submenus appear over top of the main menu, we temporarily disable focus for all sliders/buttons/etc on the
+## main menu to avoid them from grabbing keyboard focus.
+func _refresh_input_focus_mode() -> void:
+	var submenu_visible: bool = $LevelSubmenu.visible or $RegionSubmenu.visible
+	for node in get_tree().get_nodes_in_group("main_practice_inputs"):
+		node.focus_mode = FOCUS_NONE if submenu_visible else FOCUS_ALL
+
+
+## Updates the level button to show the current level's title.
+func _refresh_level_button() -> void:
+	if _region is CareerRegion:
+		_level_button.text = "%s: %s" % [_region.name, _level_settings.title]
+	else:
+		if _region.has_flag(OtherRegion.FLAG_TRAINING):
+			_level_button.text = _level_settings.title
+		else:
+			_level_button.text = "%s: %s" % [_region.branch_name, _level_settings.title]
+	_level_description_label.text = _level_settings.description
+
+
+## Updates the speed slider with a list of available speeds.
+##
+## For career regions, the player is constrained by the speeds of that particular region.
+##
+## For non-career region, the player can only speed levels up. They cannot slow them down.
+##
+## For rank regions, the player can not make any changes at all.
+func _refresh_speed_selector() -> void:
+	if _region is CareerRegion:
+		# Constrain the speed selector to region's min/max speeds
+		var min_index := PieceSpeeds.speed_ids.find(_region.min_piece_speed)
+		var max_index := PieceSpeeds.speed_ids.find(_region.max_piece_speed)
+		var new_speed_names := PieceSpeeds.speed_ids.slice(min_index, max_index)
+		_speed_selector.set_speed_names(new_speed_names)
+		_speed_selector.set_editable(true)
+	else:
+		# Constrain the speed selector to level's speeds. For most levels, we let the player choose faster speeds
+		# within a threshold.
+		var start_speed := _level_settings.speed.get_start_speed()
+		var speed_matrix_row: Array = []
+		var speed_editable := true
+		for next_speed_matrix_row in _speed_matrix:
+			if next_speed_matrix_row.has(start_speed):
+				speed_matrix_row = next_speed_matrix_row.duplicate()
+				break
+		
+		if _region.id == OtherRegion.ID_SANDBOX:
+			# Players can play sandbox modes at whatever speed they like.
+			pass
+		else:
+			# Player can't slow down training modes, they can only speed them up. This prevents them from clearing
+			# hard training levels by slowing them down
+			speed_matrix_row = speed_matrix_row.slice(speed_matrix_row.find(start_speed), speed_matrix_row.size() - 1)
+		
+		if _region.id == OtherRegion.ID_RANK:
+			# Players can't change the speed in rank mode.
+			speed_editable = false
+		
+		_speed_selector.set_speed_names(speed_matrix_row)
+		_speed_selector.set_selected_speed(start_speed)
+		_speed_selector.set_editable(speed_editable)
```

Comment:
This code could be more symmetrical, why not pull this piece out of the `if/else` block

Reply:
Refactored code, pulling the variables and assignments outside of the if block